File: 01/flavien/AoC/index-2.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final input = await File('./input.txt').readAsString();
  final reduced = LineSplitter().convert(input).map((line) {
    final lineDigits =
        RegExp(r'(?=(one|two|three|four|five|six|seven|eight|nine|\d))')
            .allMatches(line)
            .map((match) => match.group(1))
            .map((n) {
      return n == null
          ? 0
          : int.tryParse(n) == null
              ? [
                    "one",
                    "two",
                    "three",
                    "four",
                    "five",
                    "six",
                    "seven",
                    "eight",
                    "nine",
                  ].indexOf(n) +
                  1
              : int.parse(n);
    });

    return int.parse('${lineDigits.first}${lineDigits.last}');
  }).fold(0, (accumulator, currentValue) => accumulator + currentValue);

  print(reduced);
}

File: 01/flavien/AoC/index-2.mjs
import { readFile } from "fs/promises"
console.log(
  (await readFile("./input.txt", { encoding: "utf8" }))
    .split("\n")
    .map((line) => {
      const lineDigits = Array.from(
        line.matchAll(
          new RegExp(/(?=(one|two|three|four|five|six|seven|eight|nine|\d))/g)
        ),
        (m) => m[1]
      ).map((nAsNumberOrText) =>
        Number.isNaN(Number(nAsNumberOrText))
          ? [
              "one",
              "two",
              "three",
              "four",
              "five",
              "six",
              "seven",
              "eight",
              "nine",
            ].indexOf(nAsNumberOrText) + 1
          : nAsNumberOrText
      )

      return Number(`${lineDigits.at(0)}${lineDigits.at(-1)}`)
    })
    .reduce((accumulator, currentValue) => (accumulator += currentValue), 0)
)
